.github/workflows/pre-post-release.yml

```diff
@@ -0,0 +1,97 @@
+---
+name: Pre/Post Release
+
+on:
+  workflow_call:
+    ref:
+      description: 'Branch or tag ref to run the workflow on'
+      required: true
+      default: 'main'
+    version:
+      description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+      required: true
+    phase:
+      description: 'Pre or post release phase'
+      type: choice
+      options:
+        - pre
+        - post
+      required: true
+
+env:
+  RELEASE_VERSION: ${{ inputs.version }}
+  BRANCH_NAME: ${{ inputs.phase }}-release-v${{ inputs.version }}
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Validate tag does not exist on current commit
+        uses: ./.github/workflows/validate-tag
+        with:
+          tag: ${{ env.RELEASE_VERSION }}
```

Comment:
The tag in the git repo is `v${{ env.RELEASE_VERSION }}` with the extra `v` prefix.

Reply:
Ref: https://github.com/elastic/ecs-logging-java/pull/226/commits/7c2921734ae4426fbe693b57f462e2a1d8d408c5